lib/utils/BaseReporter.js

```diff
@@ -63,45 +66,76 @@ class BaseReporter extends events.EventEmitter {
         })
 
         this.on('runner:start', (runner) => {
-            if (!this.stats.runner[runner.cid]) {
-                this.stats.runner[runner.cid] = {
-                    start: new Date(),
-                    capabilities: runner.capabilities,
-                    config: runner.config,
-                    tests: []
-                }
-            }
+            this.getSpecStats(runner)
```

Comment:
@georgecrawford from what I see is that the function sets some default properties to the payload and enhances it with additional data. The confusing part is why does it return something if you actually don't assign it to any object? I am also not a fan of super long object notations like `this.stats.runner[obj.cid].specs[obj.specHash]` , it is super dangerous if one of the values is accidentally undefined. I tried to rewrite the function but didn't found a right solution. Maybe you know better what the actual purpose of it is. Maybe create a helper class to abstract this better where you can initiate an instance with a bunch of data an get a formatted object back to propagate to other reporter. What do you think?